packages/ember-glimmer/tests/integration/components/curly-components-test.js

```diff
@@ -2782,6 +2782,54 @@ moduleFor('Components test: curly components', class extends RenderingTest {
     }, /didInitAttrs called/);
   }
 
+  ['@test did{Init,Receive}Attrs fires after .init() but before finishChains'](assert) {
+    expectDeprecation(/didInitAttrs called/);
+
+    let fooDidChangeCount = 0;
+    let barDidChangeCount = 0;
+
+    let didInit = false;
+
+    let ComponentClass = Component.extend({
+      init() {
+        didInit = true;
+        this._super(...arguments);
+      },
+
+      didInitAttrs() {
+        assert.ok(didInit, 'expected init to have run before didInitAttrs');
+        this.set('foo', 1);
+      },
+
+      willReceieveAttrs() {
+        assert.ok(didInit, 'expected init to have run before willReceieveAttrs');
+        this.set('bar', 1);
+      },
+
+      fooDidChange: observer('foo', () => { fooDidChangeCount++; }),
+      barDidChange: observer('bar', () => { barDidChangeCount++; })
+    });
+
+    assert.equal(fooDidChangeCount, 0);
+    assert.equal(barDidChangeCount, 0);
+
+    let component = ComponentClass.create({ renderer: {} });
```

Comment:
the test should be tested via manual creation, which file is better suited for it. I will move this test there.